src/main/java/thePackmaster/relics/PMBoosterBox.java

```diff
@@ -1,13 +1,159 @@
+import com.evacipated.cardcrawl.mod.stslib.patches.CenterGridCardSelectScreen;
+import com.megacrit.cardcrawl.cards.CardGroup;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.helpers.PowerTip;
+import com.megacrit.cardcrawl.rewards.RewardItem;
+import com.megacrit.cardcrawl.rooms.AbstractRoom;
+import thePackmaster.SpireAnniversary5Mod;
+import thePackmaster.packs.AbstractCardPack;
 
-import static thePackmaster.SpireAnniversary5Mod.makeID;
+import java.util.ArrayList;
+import java.util.Collections;
+
+import static thePackmaster.SpireAnniversary5Mod.*;
+
+public class PMBoosterBox extends AbstractPackmasterRelic implements CustomSavable<ArrayList<String>> {
 
-public class PMBoosterBox extends AbstractPackmasterRelic {
     public static final String ID = makeID("PMBoosterBox");
 
+    int numPicked = 0;
+    private boolean cardSelected = true;
+    private String myPackOne = "";
+    private String myPackTwo = "";
+    private String myPackThree = "";
+    private ArrayList<String> myPacks = new ArrayList<>();
+    private static boolean allDone = false;
+
+
     public PMBoosterBox() {
         super(ID, RelicTier.RARE, LandingSound.FLAT);
+        allDone = false;
+    }
+
+    public void onVictory() {
+        RewardItem r = new RewardItem();
+        r.cards = getCardsFromPacks(myPacks, 3);
+        AbstractDungeon.getCurrRoom().rewards.add(new RewardItem());
+    }
+
+    @Override
+    public void onEquip() {
+        allDone = false;
+        openScreen();
+    }
+
+    @Override
+    public ArrayList<String> onSave() {
+        ArrayList<String> myList = new ArrayList<>();
+        myList.add(myPackOne);
+        myList.add(myPackTwo);
+        myList.add(myPackThree);
+        //  myList.add(myPackFour);
+        return myList;
+    }
+
+    @Override
+    public void onLoad(ArrayList<String> abstractCards) {
+        myPackOne = abstractCards.get(0);
+        myPackTwo = abstractCards.get(1);
+        myPackThree = abstractCards.get(2);
+        if (AbstractDungeon.player != null) {
+            setDescriptionAfterLoading();
+        }
+        makeIDArray();
+    }
+
+    public void makeIDArray(){
+        myPacks.clear();
+        myPacks.add(myPackOne);
+        myPacks.add(myPackTwo);
+        myPacks.add(myPackThree);
     }
+
+    private void openScreen() {
+        this.cardSelected = false;
+        if (AbstractDungeon.isScreenUp) {
+            AbstractDungeon.dynamicBanner.hide();
+            AbstractDungeon.overlayMenu.cancelButton.hide();
+            AbstractDungeon.previousScreen = AbstractDungeon.screen;
+        }
+
+        AbstractDungeon.getCurrRoom().phase = AbstractRoom.RoomPhase.INCOMPLETE;
+        CenterGridCardSelectScreen.centerGridSelect = true;
+        CardGroup group = new CardGroup(CardGroup.CardGroupType.UNSPECIFIED);
+        ArrayList<AbstractCardPack> packs = SpireAnniversary5Mod.allPacks;
+        if (myPackOne != "" && myPackOne != null){
```

Comment:
Normally you check != null before checking if something is empty, don't think this can cause issues here, just somethigns to look out for